src/sortStudents.ts

```diff
@@ -1,16 +1,52 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: [];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy: Student[] = [...students];
+  const getMark = (
+    grades: number[],
+  ): number => grades.reduce((a, b) => a + b, 0) / grades.length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return order === 'asc'
+        ? studentsCopy.sort((a, b) => a[sortBy].localeCompare(b[sortBy]))
+        : studentsCopy.sort((a, b) => b[sortBy].localeCompare(a[sortBy]));
+
+    case SortType.Age:
+    case SortType.Married:
+      return order === 'asc'
+        ? studentsCopy.sort((a, b) => +a[sortBy] - +b[sortBy])
+        : studentsCopy.sort((a, b) => +b[sortBy] - +a[sortBy]);
+
+    case SortType.AverageGrade:
+      return order === 'asc'
+        ? studentsCopy.sort((a, b) => getMark(a[sortBy]) - getMark(b[sortBy]))
+        : studentsCopy.sort((a, b) => getMark(b[sortBy]) - getMark(a[sortBy]));
```

Comment:
Follow DRY , you can return imideatly
```suggestion
return studentsCopy.sort((a, b) => {
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      return order === 'asc'
        ? studentsCopy.sort((a, b) => a[sortBy].localeCompare(b[sortBy]))
        : studentsCopy.sort((a, b) => b[sortBy].localeCompare(a[sortBy]));

    case SortType.Age:
    case SortType.Married:
      return order === 'asc'
        ? +a[sortBy] - +b[sortBy]
        :  +b[sortBy] - +a[sortBy]);

    case SortType.AverageGrade:
      return order === 'asc'
        ? getMark(a[sortBy]) - getMark(b[sortBy])
        :  getMark(b[sortBy]) - getMark(a[sortBy]);
```